gpt4all-backend/llamamodel.cpp

```diff
@@ -28,14 +28,23 @@
 #include <llama.h>
 #include <ggml.h>
 
+
 namespace {
 const char *modelType_ = "LLaMA";
 }
 
 struct gpt_params {
     int32_t seed          = -1;   // RNG seed
-    int32_t n_parts       = -1;   // amount of model parts (-1 = determine from model dimensions)
     int32_t n_keep        = 0;    // number of tokens to keep from initial prompt
+#if LLAMA_DATE <= 230511
+    int32_t n_parts       = -1;   // amount of model parts (-1 = determine from model dimensions)
+#endif
```

Comment:
The crux of it. We're going to use macros...

Reply:
Our other option would be to have an extensive collection of almost-identical `llamamodel.cpp` files for different llama.cpp versions.

Reply:
No, I think this is the right choice of a bunch of bad choices.

Reply:
There's also CRTP and C++ template magic, but I agree it's not the time to go there yet.